### PR TITLE
Formspec: Unify argument checks

### PR DESCRIPTION
**Pull request by SmallJoker**
Features:
 * Combined argument range checks to avoid forwards-compatibility mistakes
 * Additional `m_client` checks where necessary (low priority)
 * Error message corrections where it stated wrong element names

Based on the idea of v-rob: https://github.com/minetest/minetest/pull/11821#discussion_r765239200

## To do

This PR is Ready for Review.

## How to test

1. Main menu must render properly
2. devtest `/test_formspecs`
3. Code review: **Ignore space changes**: `git diff -w`  (only few lines were actually modified)

### CODE REVIEW

File: src/gui/guiFormSpecMenu.cpp [hunk 1]
```diff
@@ -445,105 +459,95 @@ void GUIFormSpecMenu::parseScrollContainerEnd(parserData *data)
 
 void GUIFormSpecMenu::parseList(parserData *data, const std::string &element)
 {
-	if (m_client == 0) {
-		warningstream<<"invalid use of 'list' with m_client==0"<<std::endl;
-		return;
-	}
+	MY_CHECKCLIENT("list");
```
Comment (sfan5) [inline src/gui/guiFormSpecMenu.cpp:462]: Can't this check go elsewhere?
The client reference being undefined seems like such an obvious error that it should even get this far.

Comment (SmallJoker) [reply src/gui/guiFormSpecMenu.cpp:462]: So I thought too until I noticed that the Main Menu uses GUIFormSpec for rendering.

Comment (sfan5) [reply src/gui/guiFormSpecMenu.cpp:462]: That's unfortunate.

**Review by sfan5 — APPROVED** [review]
Looks correct.

**Comment by nerzhul** [comment]
clearly a good refacto

**Pull request merged**